### PR TITLE
Pruning Command Line Flags

### PR DESCRIPTION
**Pull request by RatanRSur**
## PR description
Adds pruning and pruning configuration dependent flags

## Fixed Issue(s)

https://pegasys1.atlassian.net/secure/RapidBoard.jspa?rapidView=28&modal=detail&selectedIssue=PAN-2878&assignee=5ce472e018de8f0dc3552e9c

### CODE REVIEW

File: pantheon/src/main/java/tech/pegasys/pantheon/Runner.java [hunk 1]
```diff
@@ -76,9 +76,7 @@
   public void start() {
     try {
       LOG.info("Starting Ethereum main loop ... ");
-      if (natManager.isPresent()) {
-        natManager.get().start();
-      }
+      natManager.ifPresent(UpnpNatManager::start);
```
Comment (abdelhamidbakhta) [inline pantheon/src/main/java/tech/pegasys/pantheon/Runner.java]: nice catch

File: pantheon/src/main/java/tech/pegasys/pantheon/Runner.java [hunk 2]
```diff
@@ -76,9 +76,7 @@
   public void start() {
     try {
       LOG.info("Starting Ethereum main loop ... ");
-      if (natManager.isPresent()) {
```
Comment (shemnon) [inline pantheon/src/main/java/tech/pegasys/pantheon/Runner.java]: Can this go in a different PR?  It's unrelated.

File: pantheon/src/main/java/tech/pegasys/pantheon/cli/DefaultCommandValues.java [hunk 3]
```diff
@@ -43,6 +43,8 @@
   String MANDATORY_NETWORK_FORMAT_HELP = "<NETWORK>";
   String MANDATORY_NODE_ID_FORMAT_HELP = "<NODEID>";
   Wei DEFAULT_MIN_TRANSACTION_GAS_PRICE = Wei.of(1000);
+  long DEFAULT_RETENTION_PERIOD = 1000;
```
Comment (shemnon) [inline pantheon/src/main/java/tech/pegasys/pantheon/cli/DefaultCommandValues.java]: Should we up this to be at least 1024?  Maybe more?  I would like the default to be higher to ensure sweep doesn't cut past the finality gadget horizon, which starts at a 1024 block minimum.  In a live system it will be functionally more but in a static replay we may get in trouble.

Comment (RatanRSur) [reply pantheon/src/main/java/tech/pegasys/pantheon/cli/DefaultCommandValues.java]: How about we up it to 1024 for now and then see what the state size hovers around when we do our tests to see if we want to increase it further?
